xarray/tests/test_dataset.py

```diff
@@ -5607,6 +5607,17 @@ def test_pad(self):
         np.testing.assert_equal(padded["var1"].isel(dim2=[0, -1]).data, 42)
         np.testing.assert_equal(padded["dim2"][[0, -1]].data, np.nan)
 
+    def test_astype_attrs(self):
+        data = create_test_data(seed=123)
+        data.attrs["foo"] = "bar"
+
+        assert list(data.attrs.items()) == list(data.astype(float).attrs.items())
```

Comment:
you can compare dictionaries with `==` so this could be `assert data.attrs == data.astype(float).attrs`